packages/jest-expect/src/types.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import type {AsymmetricMatchers, BaseExpect, Matchers} from 'expect';
+import type {SnapshotState, addSerializer} from 'jest-snapshot';
+
+export type JestExpect = {
+  <T = unknown>(actual: T): JestMatchers<void, T> &
+    Inverse<JestMatchers<void, T>> &
+    PromiseMatchers<T>;
+  addSnapshotSerializer: typeof addSerializer;
+} & BaseExpect &
+  AsymmetricMatchers &
+  Inverse<Omit<AsymmetricMatchers, 'any' | 'anything'>>;
+
+type Inverse<Matchers> = {
+  /**
+   * Inverse next matcher. If you know how to test something, `.not` lets you test its opposite.
+   */
+  not: Matchers;
+};
+
+type JestMatchers<R extends void | Promise<void>, T> = Matchers<R> &
+  SnapshotMatchers<R, T>;
```

Comment:
Some repetition in this file. Cannot convince TS that it is fine to have unused generic argument. TS keeps on complaining about `Matchers<R, T>`, because `T` is not used in the `Matchers` interface. In a way I like to keep `Matchers` in `expect` just with one arg, but this results in some repetition here.

Reply:
repetition is fine as long as TS complains if stuff goes out of sync 🙂